spec/models/label_spec.rb

```diff
@@ -89,5 +89,115 @@
         expect(account.reload.level).to eq 0
       end
     end
+
+    context 'document label changes' do
+      let!(:account) { create(:account) }
+      let(:mailer_deliveries) { ActionMailer::Base.deliveries }
+      let!(:document_label) do
+        create(
+          :label,
+          account: account,
+          key: 'document',
+          value: 'pending',
+          scope: 'private'
+        )
+      end
+
+      context 'when label value is verified' do
+        it 'sends notification email' do
+          document_label.update(value: 'verified')
+          expect(mailer_deliveries.last.to).to eq [account.email]
+          expect(mailer_deliveries.last.subject).to eq 'Your identity was approved'
+        end
+      end
+
+      context 'when label value is rejected' do
+        it 'sends notification email' do
+          document_label.update(value: 'rejected')
+          expect(mailer_deliveries.last.to).to eq [account.email]
+          expect(mailer_deliveries.last.subject).to eq 'Your identity was rejected'
+        end
+      end
+    end
+
+    context '2 labels with same keys' do
+      let!(:account) { create(:account) }
+      let!(:label_public) do
+        create :label,
+               account: account,
+               key: email_verified_level.key,
+               value: email_verified_level.value,
+               scope: 'public'
+      end
+
+      let!(:label_private) do
+        create :label,
+               account: account,
+               key: email_verified_level.key,
+               value: email_verified_level.value,
+               scope: 'private'
+      end
+
+      context 'can be created with different scopes' do
+        it 'account has both labels' do
+          expect(account.labels.first).to eq label_public
+          expect(account.labels.second).to eq label_private
+        end
+
+        it 'account has level 1' do
+          expect(account.level).to eq 1
+        end
+
+        context 'when private label changes' do
+          it 'account level downgrades when value changed' do
+            expect { label_private.update(value: 'rejected') }.to change { account.reload.level }.to 0
+          end
+
+          it 'account level downgrades when key changed' do
+            expect { label_private.update(key: 'email0') }.to change { account.reload.level }.to 0
+          end
+        end
+
+        context 'when public label changes' do
+          it 'account level does not change' do
+            expect { label_public.update(value: 'rejected') }.to_not change { account.reload.level }
+          end
+        end
+      end
+    end
+
+    context 'with public labels duplicating levels' do
```

Comment:
Wrong name. It's not implicit reason for spec.
I suggest `when account has all required label for level 4 but in public scope`